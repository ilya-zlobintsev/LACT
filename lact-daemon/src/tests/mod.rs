use insta::assert_json_snapshot;

use crate::{config::Config, server::handler::Handler};
use std::{fs, path::PathBuf};

#[tokio::test]
async fn snapshot_everything() {
    let test_data_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src/tests/data");

    for vendor_dir in fs::read_dir(test_data_dir).unwrap().flatten() {
        for device_dir in fs::read_dir(vendor_dir.path()).unwrap().flatten() {
            let test_key = format!(
                "{}/{}",
                vendor_dir.file_name().to_string_lossy(),
                device_dir.file_name().to_string_lossy()
            );

            let handler = Handler::with_base_path(&device_dir.path(), Config::default(), true)
                .await
                .unwrap();
            let mut device_info = handler
                .generate_snapshot_device_info()
                .into_values()
                .next()
                .unwrap();
            // Remove vulkan information, as it can only be fetched from the current running system and affects snapshot output
            device_info
                .get_mut("info")
                .unwrap()
                .as_object_mut()
                .unwrap()
                .remove("vulkan_info");

            assert_json_snapshot!(test_key, device_info);
        }
    }
}
