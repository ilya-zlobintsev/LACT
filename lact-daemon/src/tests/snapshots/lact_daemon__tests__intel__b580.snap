---
source: lact-daemon/src/tests/mod.rs
expression: device_info
---
{
  "clocks_info": {
    "table": {
      "type": "intel",
      "value": {
        "gt_freq": [
          1200,
          2850
        ],
        "rp0_freq": 2850,
        "rpe_freq": 400,
        "rpn_freq": 400
      }
    }
  },
  "info": {
    "driver": "xe",
    "drm_info": {
      "memory_info": {
        "cpu_accessible_total": 0,
        "cpu_accessible_used": 0,
        "resizeable_bar": true
      },
      "vram_clock_ratio": 1.0
    },
    "link_info": {},
    "pci_info": {
      "device_pci_info": {
        "model": "Battlemage G21 [Arc B580]",
        "model_id": "E20B",
        "vendor": "Intel Corporation",
        "vendor_id": "8086"
      },
      "subsystem_pci_info": {
        "model": "Intel Arc B580 Limited Edition",
        "model_id": "1100",
        "vendor": "Intel Corporation",
        "vendor_id": "8086"
      }
    },
    "vulkan_instances": []
  },
  "pci_info": {
    "device_pci_info": {
      "model": "Battlemage G21 [Arc B580]",
      "model_id": "E20B",
      "vendor": "Intel Corporation",
      "vendor_id": "8086"
    },
    "subsystem_pci_info": {
      "model": "Intel Arc B580 Limited Edition",
      "model_id": "1100",
      "vendor": "Intel Corporation",
      "vendor_id": "8086"
    }
  },
  "power_profile_modes": null,
  "power_states": {
    "core": [
      {
        "enabled": true,
        "value": 400
      },
      {
        "enabled": true,
        "value": 400
      },
      {
        "enabled": true,
        "value": 2850
      }
    ],
    "vram": []
  },
  "stats": {
    "clockspeed": {
      "current_gfxclk": 2850,
      "gpu_clockspeed": 2733
    },
    "fan": {
      "control_enabled": false,
      "pmfw_info": {},
      "speed_current": 1292
    },
    "power": {
      "cap_current": 180.0,
      "cap_default": 180.0,
      "cap_max": 380.0,
      "cap_min": 0.0
    },
    "temps": {
      "pkg": {
        "crit": null,
        "crit_hyst": null,
        "current": 62.0
      },
      "vram": {
        "crit": null,
        "crit_hyst": null,
        "current": 64.0
      }
    },
    "throttle_info": {
      "pl2": []
    },
    "voltage": {},
    "vram": {}
  }
}
